### PR TITLE
Try suggested fixes

### PR DESCRIPTION
**Pull request by dansanduleac**
Apply the suggestions I made to https://github.com/palantir/gradle-revapi/pull/221

### CODE REVIEW

File: src/main/java/com/palantir/gradle/revapi/RevapiAnalyzeTask.java [hunk 1]
```diff
@@ -126,16 +131,15 @@ private RevapiConfig revapiIgnores() {
...
                 .addArchives(toFileArchives(apiJars))
                 .addSupportArchives(toFileArchives(dependencyJars))
                 .build();
     }
 
-    private static List<FileArchive> toFileArchives(SetProperty<File> property) {
-        return property.get().stream()
-                .filter(File::isFile)
+    private static List<FileArchive> toFileArchives(Provider<FileCollection> property) {
+        return property.get().filter(File::isFile).getFiles().stream()
```
Comment (dansanduleac) [inline src/main/java/com/palantir/gradle/revapi/RevapiAnalyzeTask.java:142]: I'd actually just throw here if you get smth that's not a file. We control what goes in here, for the most part, and if smth that's not a file ends up in here, that's bad and we should alert the user that we're not analysing their entire configuration.

Comment (dansanduleac) [reply src/main/java/com/palantir/gradle/revapi/RevapiAnalyzeTask.java:142]: well, never mind, the `handles the output of extra source sets being added to compile configuration` test promptly told me why we can't just do that.

File: src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy [hunk 2]
```diff
@@ -749,6 +751,16 @@ class RevapiSpec extends IntegrationSpec {
...
+ - The configuration :api-jersey:compileClasspath was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.6.4/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.
+ - The configuration :api-retrofit:compileClasspath was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a thread not managed by Gradle or from a different project.  See https://docs.gradle.org/5.6.4/userguide/troubleshooting_dependency_resolution.html#sub:configuration_resolution_constraints for more details. This behaviour has been deprecated and is scheduled to be removed in Gradle 6.0.
+        */
+        System.setProperty('ignoreMutableProjectStateWarnings', 'true')
+        System.setProperty('ignoreDeprecations', 'true')
```
Comment (dansanduleac) [inline src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:763]: I don't know why this was passing on circle but it was failing when run from IDEA, so had to make it shut up...

Comment (CRogers) [reply src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:763]: I don't believe we test again Gradle 6 in this repo	- does this work for sure in Gradle 6?

Comment (CRogers) [reply src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy:763]: Looks like it does work in Gradle 6.2.2

**Comment by CRogers** [comment]
:+1:

**Pull request merged**